### ISSUE TITLE
graph-tool: Could not link test program to Python

### ISSUE THREAD

**Issue by shongololo** [Homebrew/homebrew-science#6461]
### Indicate troubleshooting steps taken

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Checked that a similar issue hasn't already been filed?

### Bug reports / build errors:
```
Could not link test program to Python. Maybe the main Python library has been
  installed in some non-standard library path. If so, pass it to configure,
  via the LIBS environment variable.
  Example: ./configure LIBS="-L/usr/non-standard-path/python/lib"
  ============================================================================
   ERROR!
   You probably have to install the development version of the Python package
   for your distribution.  The exact name of this package varies among them.
  ============================================================================

```
#### Brief summary of issue:
The standard `graph-tool` package installs fine, but when using the `--with-python3` flag it throws the above error.

#### The complete command that you issued:
Various permutations:
```
brew install -v --build-from-source --with-python3 --without-python graph-tool
brew install -v --with-python3 graph-tool
brew install -v --with-python3 --without-python graph-tool
```

#### Gist with output of `brew doctor`:
`Your system is ready to brew.`

#### Link output by `brew gist-logs <formula>`:
https://gist.github.com/shongololo/a940264f99dc6e974c8526d96343c420

#### Output of `brew config`:
```
HOMEBREW_VERSION: 1.3.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: 441c212f2dd70f053ee21eac8398c5fe5dfd0551
Last commit: 10 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 989e70179390f67c0ddfc36f7289fbb20b1b3c65
Core tap last commit: 3 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_GIT_CONFIG_FILE: /usr/local/Homebrew/.git/config
CPU: octa-core 64-bit skylake
Homebrew Ruby: 2.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby
Clang: 9.0 build 900
Git: 2.8.1 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Perl: /usr/bin/perl
Python: /usr/local/opt/python/libexec/bin/python => /usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/local/bin/ruby => /usr/local/Cellar/ruby/2.4.2_1/bin/ruby
Java: 1.8.0_101
macOS: 10.13.1-x86_64
Xcode: N/A
CLT: 9.1.0.0.1.1508540944
X11: 2.7.11 => /opt/X11
```

**Comment by count0** [comment]
The `configure.log` you linked to does not show any error, and points to python 2.7. Could you please send the logs to the corresponding error?

**Comment by shongololo** [comment]
@count0 is it possible that the `--with-python3` flag isn't properly registering with the build script? i.e. that the build script is trying to test against python 2.7 even though the `--without-python --with-python3` flags are passed?

This is the install command that I'm running:
`brew install graph-tool --without-python --with-python3`

Which gives this error:
```
checking consistency of all components of python development environment... no
configure: error: in `/private/tmp/graph-tool-20171122-57584-1rk0wse/graph-tool-2.26/build-python3-3.6':
configure: error: 
  Could not link test program to Python. Maybe the main Python library has been
  installed in some non-standard library path. If so, pass it to configure,
  via the LIBS environment variable.
  Example: ./configure LIBS="-L/usr/non-standard-path/python/lib"
  ============================================================================
   ERROR!
   You probably have to install the development version of the Python package
   for your distribution.  The exact name of this package varies among them.
  ============================================================================

See `config.log' for more details
```

And when I run `brew gist-logs graph-tool` it gives me this output:
https://gist.github.com/bfac2742fbff6e0443e6ff0aa688a4b8

**Comment by shongololo** [comment]
@count0 any possibility of updates on what the issue might be? Please let me know if you need further information from my end. Thanks.

**Comment by shongololo** [comment]
@count0  the only files I can find are in the `.../Logs/Homebrew/graph-tool` folder, let me know if I should paste any of these:
`03.configure.cc`
`03.configure`
`02.make.cc`
`02.make`
`01.configure.cc`
`01.configure`
`00.options.out`

I believe these are the same ones

**Comment by count0** [comment]
The `config.log` file is to be found in the build directory, usually in `/tmp/graph-tool-XXX` or `~/Library/Caches/Homebrew`.

**Comment by shongololo** [comment]
Hi @count0 - I tried looking in those directories and searching from the command line but couldn't find any graph-tool logs.

I then tried rerunning the install command with an explicit `HOMEBREW_LOGS` environment variable per:
`export HOMEBREW_LOGS=~/Desktop/tmp-homebrew`
But the files outputted to the logging directory are the same ones that get published with brew's `gist-logs` command:
```
03.configure.cc
03.configure
02.make.cc
02.make
01.configure.cc
01.configure
00.options.out
```
Here is the gist log for the latest attempt:
https://gist.github.com/anonymous/9a91a82638c3803d896406d1c2df55c4

For what it is worth I get this error on two separate computers.

**Comment by shongololo** [comment]
@count0 The log files are now showing up, presumably something to do with the `--keep-tmp` flag.

I've attached the two log files, one for Python 2.7 (which exits with code 0) and one for Python 3.6 (which exits with code 1).

[config_py27.log](https://github.com/Homebrew/homebrew-science/files/1504055/config_py27.log)
[config_py36.log](https://github.com/Homebrew/homebrew-science/files/1504057/config_py36.log)

**Comment by shongololo** [comment]
@count0 I've tried on a third computer with a completely fresh brew install and the same issue arises. The log file looks similar to the one posted above but I can attach if a separate example is beneficial?

**Comment by count0** [comment]
What is the output of the following command in your system?
```
python3 -c "import distutils.sysconfig; conf = distutils.sysconfig.get_config_var; print (conf('LIBS') + ' ' + conf('SYSLIBS'))"
```

**Comment by shongololo** [comment]
This is what it prints:
`-ldl  -framework CoreFoundation `

**Comment by count0** [comment]
How about:
```
python3 -c "import distutils.sysconfig; conf = distutils.sysconfig.get_config_var; print(conf('LINKFORSHARED'))"
```

**Comment by shongololo** [comment]
`-Wl,-stack_size,1000000  -framework CoreFoundation Python.framework/Versions/3.6/Python`

**Comment by count0** [comment]
It looks like this is a regression of: https://github.com/Homebrew/legacy-homebrew/issues/37923

I'll open a new issue.

**Comment by sjackman** [comment]
`graph-tool` is up for migration to Homebrew/core. Consider submitting this formula to https://github.com/Homebrew/homebrew-core. This repository Homebrew/science is being retired at the end of this year. See https://github.com/Homebrew/homebrew-science/issues/6331

Contributing this formula to Homebrew/core would also build a binary bottle for it, which would likely resolve your issue.

Here's the top twelve installation options:
```
 1 | homebrew/science/graph-tool                               | 1,393 |  65.93%
 2 | homebrew/science/graph-tool --with-python3                |   250 |  11.83%
 3 | homebrew/science/graph-tool --without-python --with-pytho |   126 |   5.96%
 4 | homebrew/science/graph-tool --without-matplotlib --withou |    76 |   3.60%
 5 | homebrew/science/graph-tool --with-openmp                 |    39 |   1.85%
 6 | homebrew/science/graph-tool --with-openmp --with-python3  |    27 |   1.28%
 7 | homebrew/science/graph-tool --without-matplotlib --withou |    27 |   1.28%
 8 | homebrew/science/graph-tool --without-matplotlib --withou |    25 |   1.18%
 9 | homebrew/science/graph-tool --without-python --with-openm |    24 |   1.14%
10 | homebrew/science/graph-tool --without-matplotlib --with-o |    19 |   0.90%
11 | homebrew/science/graph-tool --without-cairo --without-gtk |    18 |   0.85%
12 | homebrew/science/graph-tool --HEAD                        |    16 |   0.76%
```

**Comment by shongololo** [comment]
The issue has now per resolved per #21212
